app/controllers/articles_controller.rb

```diff
@@ -244,6 +243,16 @@ def set_article
     @article = found_article || not_found
   end
 
+  def set_surrogate_keys_by_case
```

Comment:
I don't think we should have this method at all.

We already have a `if` in the `feed` which either calls `handle_user_or_organization_feed` or `handle_tag_feed` or `@articles.where(featured: true).includes(:user)`, let's put the proper set_surrogate_key call at the end of those methods. This way we don't have to wonder where these instance variables come from. For example:

```ruby
  def handle_tag_feed
    @tag = Tag.aliased_name(params[:tag])
    return unless @tag

    @articles = @articles.cached_tagged_with(@tag)
    
    set_surrogate_key_header @tag_model.record_key, Article.table_key, Tag.table_key, @articles.map(&:record_key)
  end
```